SAP/fsm-shell
# [CPB-65865][SHELL] - enhance the REQUIRE_CONTEXT event to retrieve th…

…e extension deployment id

How to test:

Checkout PR (feature/CPB-65865) for fsm-shell
Go to fsm-shell
Execute npm i
Execute npm run package

Checkout feature/CPB-65865 for portal
Go to portal\packages\outlets
Execute npm i
Execute npm link ../../../fsm-shell
Execute npm run package

Go to \portal\shell\frontend
Execute npm i
Execute npm link ../../../fsm-shell

Go to \portal\shell\frontend\projects\integration-test
Execute npm link ../../../../../fsm-shell
Execute npm link ../../../../packages/outlets/dist/outlets

Start locally Shell and Integration Test Page
Make sure you have at least one extension installed with ShellSdk (like Station Clock)
Go to Integration Test Page, select Outlets and then choose One To One
Assign extension to one of those free outlets
Leave Configuration mode and refresh
Open Chrome developer tools and set breakpoint on line 465 for file ShellSdk.ts
Go to Integration Test Page, select Outlets and then choose One To One ==> Chrome should stop at the breakpoint. Payload should contain extensionDeploymentId.

Important:
I will split the changes for shell and the outlets package later in two different PRs. But for testing it is easier to have those changes in one branch.
